egs/wsj/s5/steps/cleanup/internal/get_pron_stats.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+
+# Copyright 2016  Xiaohui Zhang
+# Apache 2.0.
+
+from __future__ import print_function
+import argparse
+import sys
+import warnings
+
+# Collect pronounciation stats from a ctm_prons.txt file (containing entries: utt_id word phone_1...phone_N
+# which must be consecutive in time) and output the stats (format: count word pronounciation) into prons.txt
+def GetArgs():
+    parser = argparse.ArgumentParser(description = "Accumulate pronounciation statistics from "
+                                     "a ctm_prons.txt file.",
+                                     epilog = "See steps/cleanup/debug_lexicon.sh for example")
+    parser.add_argument("ctm_prons_file", metavar = "<ctm-prons-file>", type = str,
+                        help = "File containing word-pronounciation alignments obtained from a ctm file; "
+                        "each line must be <utt_id> <word> <phones>")
+    parser.add_argument("silence_file", metavar = "<silphone-file>", type = str,
+                        help = "File containing a list of silence phones.")
+    parser.add_argument("optional_silence_file", metavar = "<optional_silence>", type = str,
+                        help = "File containing the optional silence phone. We'll be replacing empty prons by this.")
+    parser.add_argument("non_scored_words_file", metavar = "<non-scored-words-file>", type = str,
+                        help = "File containing a list of non-scored words.")
+    parser.add_argument("stats_file", metavar = "<stats-file>", type = str,
+                        help = "Write accumulated statitistics to this file;"
+                        "each line is <count> <word> <phones>")
+    print (' '.join(sys.argv), file=sys.stderr)
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if args.ctm_prons_file == "-":
+        args.ctm_prons_file_handle = sys.stdin
+    else:
+        args.ctm_prons_file_handle = open(args.ctm_prons_file)
+    args.non_scored_words_file_handle = open(args.non_scored_words_file)
+    args.silence_file_handle = open(args.silence_file)
+    args.optional_silence_file_handle = open(args.optional_silence_file)
+    if args.stats_file == "-":
+        args.stats_file_handle = sys.stdout
+    else:
+        args.stats_file_handle = open(args.stats_file, "w")
+    return args
+
+def ReadEntries(file_handle):
+    entries = set()
+    for line in file_handle:
+        entries.add(line.strip())
+    return entries
+
+# Basically, this function generates an "info" list from a ctm_prons file.
+# Each entry in the list represents the pronounciation candidate(s) of a word.
+# For each non-<eps> word, the entry is a list: [utt_id, word, set(prnounciation_candidates)].
```

Comment:
typo on this line.
Also, please give an example of the data structure, e.g.
`[ 'foo-bar123', 'hello', set(['h ah l l', 'ha l l ow' ])]` or whatever it is.